handsontable/src/plugins/exportFile/types/csv.js

```diff
@@ -66,23 +83,73 @@ class Csv extends BaseType {
    * Escape cell value.
    *
    * @param {*} value Cell value.
-   * @param {boolean} [force=false] Indicates if cell value will be escaped forcefully.
+   * @param {object} options Options.
+   * @param {boolean} [options.force=false] Indicates if cell value will be escaped forcefully.
+   * @param {boolean|RegExp|Function} [options.sanitizeValue=false] Controls the sanitization of cell value.
    * @returns {string}
    */
-  _escapeCell(value, force = false) {
-    let escapedValue = stringify(value);
+  _escapeCell(value, { force = false, sanitizeValue = false } = {}) {
+    let returnValue = stringify(value);
+
+    if (returnValue === '') {
+      return returnValue;
+    }
+
+    if (sanitizeValue) {
+      force = true;
+    }
+
+    if (sanitizeValue instanceof RegExp) {
+      returnValue = this._sanitizeValueWithRegExp(returnValue, sanitizeValue);
+    } else if (typeof sanitizeValue === 'function') {
+      returnValue = sanitizeValue(returnValue);
+    } else if (sanitizeValue) {
+      returnValue = this._sanitizeValueWithOWASP(returnValue);
+    }
+
+    if (force
+      || returnValue.indexOf(CHAR_CARRIAGE_RETURN) >= 0
+      || returnValue.indexOf(CHAR_DOUBLE_QUOTES) >= 0
+      || returnValue.indexOf(CHAR_LINE_FEED) >= 0
+      || returnValue.indexOf(this.options.columnDelimiter) >= 0) {
+      returnValue = returnValue.replace(new RegExp('"', 'g'), '""');
+      returnValue = `"${returnValue}"`;
+    }
 
-    if (escapedValue !== '' && (force ||
-      escapedValue.indexOf(CHAR_CARRIAGE_RETURN) >= 0 ||
-      escapedValue.indexOf(CHAR_DOUBLE_QUOTES) >= 0 ||
-      escapedValue.indexOf(CHAR_LINE_FEED) >= 0 ||
-      escapedValue.indexOf(this.options.columnDelimiter) >= 0)) {
+    return returnValue;
+  }
 
-      escapedValue = escapedValue.replace(new RegExp('"', 'g'), '""');
-      escapedValue = `"${escapedValue}"`;
+  /**
+   * Sanitize value that may be interpreted as a formula in spreadsheet software.
+   * Following the OWASP recommendations: https://owasp.org/www-community/attacks/CSV_Injection.
+   *
+   * @param {string} value Cell value.
+   * @returns {string}
+   * @private
+   */
+  _sanitizeValueWithOWASP(value) {
```

Comment:
Not crucial, but you could utilize the private method notation with `#`, so `#sanitizeValueWithOWASP`. I think this plugin doesn't use it yet because it was created before we supported it.

Reply:
done

Reply:
@sequba I can see this change broke the test. These "private" functions don't seem to be that private 🙄 . I think you can either revert this change or try to refactor the plugin to make them actually private, whatever you think it's best.

Reply:
I changed `escapeCell` to be public again